src/providers/dataProvider/index.test.ts

```diff
@@ -104,3 +117,70 @@ describe('generateBatchId', () => {
     })
   })
 })
+
+describe('generateBatchId for values greater than 9', () => {
+  let provider: DataProvider
+
+  beforeAll(async () => {
+    provider = await getDataProvider()
+    for (let i = 0; i < 20; i++) {
+      await generateBatch(
+        i,
+        provider,
+        year,
+        await generateBatchId(provider, year)
+      )
+    }
+  })
+
+  describe('validate batch numbers', () => {
+    it('should validate first 20 batchNumber and delete first 10 entries', async () => {
+      const batchData = await provider.getList(constants.R_BATCHES, {
+        sort: { field: 'id', order: 'ASC' },
+        pagination: { page: 1, perPage: 1000 },
+        filter: { yearOfReceipt: year }
+      })
+      for (let i = 0; i < batchData.data.length; i++) {
+        expect(batchData.data[i].batchNumber).toBe(
+          `V${i.toLocaleString('en-US', {
+            minimumIntegerDigits: 2,
+            useGrouping: false
+          })}/${year}`
+        )
+      }
+      const ids = []
+      // deleting first 10 entries
+      for (let i = 0; i < 10; i++) ids.push(i)
+      await provider.deleteMany(constants.R_BATCHES, { ids })
+    })
+  })
+
```

Comment:
aah, now I see what is happening. This test relies on the outcome of the previous test (that the  10 items were deleted).

Tests should not rely on each other.

I _guess_ you could just _move_ the contents of the second test into the first test.